json_object.c

```diff
@@ -1265,3 +1265,128 @@ int json_object_equal(struct json_object* jso1, struct json_object* jso2)
 	return 0;
 }
 
+static int json_object_copy_serializer_data(struct json_object *src, struct json_object *dst)
+{
+	/* `src` and `dst` were validated by now ==> no null check for them */
+	if (src->o_type != json_type_double)
+		return 0;
+	/* Nothing to do */
+	if (!src->_userdata)
```

Comment:
_userdata doesn't actually need to be non-NULL.  You can set a serializer that doesn't need userdata.